### ISSUE TITLE
HHVM PHP7 mode

### ISSUE THREAD

**Issue by thekid** [xp-forge/mirrors#44]
There are several issues with HHVM PHP7 mode, see https://github.com/xp-framework/core/issues/176

* [x] HHVM does not support `void`, the testsuite only checks for PHP >= 7.1.0
* [x] Variadic params with type constraints are not supported in non-Hack files
* [x] Types for variadic parameters are always `array`.
* [ ] ~~Field value access via `ReflectionProperty::(get|set)Value()` is broken~~ -hrm, cannot reproduce this?!

**Comment by thekid** [comment]
```sh
 $ XP_RT=hhvm xp -m ../unittest/ test src/test/php/
# ...

♥: 891/898 run (7 skipped), 891 succeeded, 0 failed
Memory used: 8192.00 kB (8192.00 kB peak)
Time taken: 10.879 seconds
```

**Comment by thekid** [comment]
> hrm, cannot reproduce this?!

Wow:

```sh
$ XP_RT=hhvm xp -w '
  class A { public $f; }
  $r= (new ReflectionClass("A"))->getProperty("f");
  return $r->getValue(new A())
'

Fatal error: Uncaught TypeError: Argument 2 passed to hphp_get_property() must be an instance of string, null given

$ XP_RT=hhvm xp -w '
  class A { public $f= "Test"; }
  $r= (new ReflectionClass("A"))->getProperty("f");
  $r->setAccessible(true);   // This makes the difference!
  return $r->getValue(new A())
'
Test
```

**Comment by thekid** [comment]
:shipit: Fix released in [5.0.1-RELEASE](https://github.com/xp-forge/mirrors/releases/tag/v5.0.1)
